docs/public/specifications/tests/Basic-TP/basic01.md

```diff
@@ -89,169 +89,167 @@ DNS queries follow, unless otherwise specified below, what is specified for
-       2. Save the name server IP and parent zone name to the *AA CNAME Found*
-          set.
-   5.  If the response is a [Referral] with a CNAME record with *Child Zone* as
-       owner name in the answer section, then
-       1. Save the name server IP and parent zone name to the *Parent Found* set.
-       2. Save the name server IP and parent zone name to the
-          *CNAME with Referral Found* set.
-   6.  If the response has the AA flag set and an empty answer section (NODATA),
-       then
-       1. [Send] a *DNAME Query* to the name server IP address.
-       2. If there is a response with the AA flag set, the [RCODE Name] NoError
-          and a DNAME record with *Child Zone* as owner name in the answer
-          section, then
-          1. Save the name server IP and parent zone name to the *Parent Found*
-            set.
-          2. Save the name server IP, parent zone name and the DNAME target
-            (RDATA value) to the *AA DNAME Found* set.
-       3. Else (no response or some other response than above), then do:
-          1. Save the name server IP and parent zone name to the *Parent Found*
-             set.
-          2. Save the name server IP and parent zone name to the
-             *AA NODATA Found* set.
-
-8. If the *Parent Found* set is non-empty, then
+
+   1.  Extract and remove *Server Address* including its *Zone Name* from
+       *Remaining Servers*.
+   2.  Insert *Server Address* and *Zone Name* into *Handled Servers*.
+   3.  Create [DNS queries][DNS Query]:
+       1. Query type SOA and query name *Zone Name* ("Zone Name SOA Query").
+       2. Query type NS and query name *Zone Name* ("Zone Name NS Query").
+   4.  [Send] *Zone Name SOA Query* to *Server Address*.
+   5.  Output [B01_SERVER_ZONE_ERROR] with query name *Zone Name*, [query type]
+       SOA and name server IP *Server Address* and go to next server in
+       *Remaining Servers* if one or more of the following matches:
+          * No DNS response.
+          * [RCODE Name] besides NoError in response.
+          * AA bit not set in response.
+          * No SOA record in answer section
+          * Owner name of SOA record is not *Zone Name*.
+   6.  [Send] *Zone Name NS Query* to *Server Address*.
+   7.  Output [B01_SERVER_ZONE_ERROR] with query name *Zone Name*, [query type]
+       SOA and name server IP *Server Address* and go to next server in
+       *Remaining Servers* if one or more of the following matches:
+          * No DNS response.
+          * [RCODE Name] besides NoError in response.
+          * AA bit not set in response.
+          * No NS records in answer section
+          * Owner name of NS records is not *Zone Name*.
+   8.  Extract the name server names from the NS records and any address records
+       in the additional section.
+   9.  Do [DNS Lookup] of name server names (A and AAAA) not already listed in the
+       additional section of the response.
```

Comment:
What if any of those lookups fail? (Maybe other test cases catch that and report on these conditions, but if so, it might be nice to add a note.)

Reply:
Nothing to be done, but I will make it explicit.